predicate/predicate.py

```diff
@@ -99,15 +107,44 @@ def values_list(self, obj):
             direct = not field.auto_created or field.concrete
         accessor = self if direct else field.get_accessor_name()
         try:
-            result = getattr(obj, accessor)
+            return getattr(obj, accessor)
         except ObjectDoesNotExist:
-            values.append(None)
+            # Occurs in evaluating reverse OneToOneField relationships.
+            return None
+
+    def _apply_lookup(self, obj):
+        """
+        Returns the result of applying this lookup, via either:
+         - The django field accessor if defined.
+         - `getattr` if the object has the appropriate attribute.
+         - `__getitem__` if the object has a matching key (in a dict).
+        """
+        if isinstance(obj, models.Model):
+            try:
+                return self._get_django_field_lookup(obj)
+            except (FieldDoesNotExist, AttributeError):
+                pass
+        if hasattr(obj, self):
+            return getattr(obj, self)
+        elif isinstance(obj, dict):
+            try:
+                return obj[self]
+            except (KeyError, IndexError):
```

Comment:
Good catch. This is superfluous. Removed in eb4d1eff56f43783bd7abea1e2a75a7b72ea8126.